### PR TITLE
 CF/BF - OSD Slave build system

### PR DESCRIPTION
**Pull request by hydra**
First cut of allow building of the OSD SLAVE board without all the flight controller code.

Likely the dependencies can be further improved.  This is a minimal-impact solution while there are a lot of other Betaflight PRs still pending.

### CODE REVIEW

File: src/main/osd_slave/osd_slave_init.h [hunk 1]
```diff
@@ -17,8 +17,13 @@
 
 #pragma once
```
Comment (hydra) [inline src/main/osd_slave/osd_slave_init.h]: config_profile.h was deleted, not renamed - git is tracking this change incorrectly.

File: Makefile [hunk 2]
```diff
@@ -781,10 +797,20 @@ COMMON_SRC = \
...
+            
+COMMON_DEVICE_SRC = \
             $(CMSIS_SRC) \
             $(DEVICE_STDPERIPH_SRC)
 
+ifeq ($(OSD_SLAVE),yes)
+TARGET_FLAGS := -DUSE_OSD_SLAVE $(TARGET_FLAGS)
+COMMON_SRC := $(COMMON_SRC) $(OSD_SLAVE_SRC) $(COMMON_DEVICE_SRC)
+else
+TARGET_FLAGS := -DUSE_FC $(TARGET_FLAGS)
```
Comment (martinbudden) [inline Makefile]: I don't really like `USE_FC` since:
1. it is the assumed default
2. It is complementary to `USE_OSD_SLAVE`

So I'd rather see `#ifndef USE_OSD_SLAVE` than `#ifdef USE_FC` in the code.

In fact, thinking about it, if `USE_FC` is used, then, in principle, the whole codebase should be under an `USE_FC`. So I really don't like it at all.

File: src/main/fc/fc_tasks.c [hunk 3]
```diff
@@ -50,10 +49,12 @@
 #include "fc/cli.h"
 #include "fc/fc_dispatch.h"
 
+#ifdef USE_FC
```
Comment (martinbudden) [inline src/main/fc/fc_tasks.c]: I prefer not to have `#ifdef`s around `#includes` - the minimal build time saving is not worth the loss of clarity.

File: src/main/flight/pid.h [hunk 4]
```diff
@@ -85,11 +85,6 @@ typedef struct pidProfile_s {
     float rateAccelLimit;                   // accel limiter roll/pitch deg/sec/ms
 } pidProfile_t;
 
-#if FLASH_SIZE <= 128
```
Comment (martinbudden) [inline src/main/flight/pid.h]: This is an unrelated change. If we want to do this, it should be in a separate PR

File: src/main/sensors/battery.c [hunk 5]
```diff
@@ -290,11 +290,15 @@ void batteryInit(void)
             break;
 
         case CURRENT_METER_VIRTUAL:
+#ifdef USE_FC
```
Comment (martinbudden) [inline src/main/sensors/battery.c]: There should be a separate `USE_CURRENT_METER_VIRTUAL` `#define` for this.

File: src/main/fc/fc_msp.c [hunk 6]
```diff
@@ -586,6 +596,256 @@ static bool mspFcProcessOutCommand(uint8_t cmdMSP, sbuf_t *dst, mspPostProcessFn
...
+        }
+        break;
+
+    case MSP_ANALOG:
+        sbufWriteU8(dst, (uint8_t)constrain(getBatteryVoltage(), 0, 255));
+        sbufWriteU16(dst, (uint16_t)constrain(getMAhDrawn(), 0, 0xFFFF)); // milliamp hours drawn from battery
+        sbufWriteU16(dst, 0); // rssi
+        sbufWriteU16(dst, (int16_t)constrain(getAmperage(), -0x8000, 0x7FFF)); // send current in 0.01 A steps, range is -320A to 320A
+        break;
+
```
Comment (martinbudden) [inline src/main/fc/fc_msp.c]: Whitespace - let's be consistent and have a single blank line at the end of each `case`.

File: Makefile (hunk 2 above)
Comment (ledvinap) [inline Makefile]: @martinbudden : What happens when other non-fc variants are added? It can get out of hand quickly ...

Comment (mikeller) [inline Makefile]: @ledvinap: Even then I think it makes more sense to encode `USE_NON_FC` (or similar) as the special case, and not the default case.

Comment (ledvinap) [inline Makefile]: `NON_FC_TARGET` ? 
`USE_NON_FC` is a bit confusing - it will be used to omit code, not include it ..

Comment (mikeller) [inline Makefile]: Yeah, that sounds better.

Comment (martinbudden) [inline Makefile]: @ledvinap , the trouble is `USE_FC` doesn't mean "use FC" it means "not osd slave"

"What happens when other non-fc variants are added? It can get out of hand quickly ..." - well we already have another non-fc target, the simulator, and it's build flags are somewhat orthogonal to the osd slave issue.

And I certainly don't favour `USE_NON_FC` since this will result in double-negatives all over the place.

I think the correct thing to do is to use `#ifdef USE_OSD_SLAVE` and `#ifndef USE_OSD_SLAVE` as appropriate.

File: src/main/flight/pid.h (hunk 4 above)
Comment (hydra) [inline src/main/flight/pid.h]: it moved to a different file.

File: src/main/sensors/battery.c (hunk 5 above)
Comment (hydra) [inline src/main/sensors/battery.c]: ok, no problem.

File: src/main/fc/fc_msp.c (hunk 6 above)
Comment (hydra) [inline src/main/fc/fc_msp.c]: yes, 👍 I will look for other instances too.  in the msp file all the main case statements are at the wrong indent level right now too; I'll leave that issue for another discussion/PR though

File: src/main/fc/fc_tasks.c (hunk 3 above)
Comment (hydra) [inline src/main/fc/fc_tasks.c]: it wasn't to do with the build time saving, it was to do with FC specific #defines being needed.

**Comment by hydra** [comment]
thanks for the review guys, I'll update the PR.

File: Makefile (hunk 2 above)
Comment (hydra) [inline Makefile]: agreed that `USE_NON_FC` or `NON_FC_TARGET` is not good because of the double negative issue.  I deliberated on that myself for a while and settled on the two positive variations.

I'm happy to use `#ifndef USE_OSD_SLAVE` instead of `#ifdef USE_FC` though i prefer avoiding `#ifndef` in general, finding a balance is tricky.  We have similar issues with the `SKIP_*` defines.

File: src/main/fc/fc_tasks.c (hunk 3 above)
Comment (martinbudden) [inline src/main/fc/fc_tasks.c]: So just include the headers unconditionally.

File: Makefile (hunk 2 above)
Comment (martinbudden) [inline Makefile]: I thought we'd got rid of all the `SKIP_`, but, checking, there are indeed a few remaining. That should be tidied up at some point.

**Comment by hydra** [comment]
@martinbudden ok, changes made.  I'll merge to master once the build passes and create a BF PR for you.

**Comment by hydra** [comment]
@weaselpup what?
